OMCompiler/Compiler/Util/List.mo

```diff
@@ -1011,26 +1011,6 @@ algorithm
   outList := append_reverse(outList, l1);
 end mergeSorted;
 
-public function sortIntN
```

Comment:
Why did you remove this function? Is it not used anymore? Seems reasonable to have a linear sorting function if the maximum value is known.

Reply:
1) The complexity of this function is not linear on the input size (which is what I think you are implying) it is O(inN). In other words, say if you want to sort {2,1,3,1000} this 1000x in ops and memory (compared to say O(n*log(n)) = 8x). Not to mention the other non-asymptotic things like allocating such a big new array and initializing it.

2) let's say this linear. How often do you sort large lists of 'n' integers between 1 and N where n~N?

3) Do you need this function right now? Do you think it can be implemented when and if needed?

4) Is this really an important issue in this PR? Are there no other things like overall design and implementation choices that you want to address after going through it twice?

Reply:
Well, it's O(max (N,inN)). For inN = 1000 and sorting around 300 elements you break even. It should probably be called radix sort if it remained. One of the benefits of it is that no comparison is performed. If using a loop it should probably pipeline rather well especially since this version does not count number of hits.

Reply:
To be fair I discussed this with Mahder, and supported it's removal. The reasoning was that it had wrong naming, did not handle duplicates and the use of it was removed if I remember. I am also to be blamed!

Reply:
> Well, it's O(max (N,inN)). 
If n > inN the input list is not unique. In that case the function does sort + unique in O(n) time and gives back inN elements. While that is still a valid operation, I think it is not what the function was intended to do.

Reply:
As an answer to 4.: Everything else seems fine, these were the only two small things i recognized.